core/src/main/java/io/grpc/internal/WritableBuffer.java

```diff
@@ -54,5 +64,5 @@
    * Releases the buffer, indicating to the {@link WritableBufferAllocator} that
    * this buffer is no longer used and its resources can be reused.
    */
-  void release();
+  void close();
```

Comment:
Changes in this interface and MessageFramer aren't directly related to the goal of this PR. Also, I'd vote for `close()` as the mirror of `ReadableBuffer`.

Reply:
@voidzcy could you clarify which changes specifically you're talking about in each? Obviously `write(InputStream)` is part of it. The other two methods in this interface are to make it compatible with `OutputStream` so that impls can impl both if useful, which `NettyWritableBuffer` now does, used [here](https://github.com/grpc/grpc-java/pull/7369/files#diff-d2ac726c357eecf72bc7927af4b67665R65).

Reply:
I mean changes like s/release/close/g, casting `int` to `byte` as well as adding the `len > 0` check to the existing write-to-array method in MessageFramer,  are sort of excessive regarding to the goal of this PR. They are irrelevant to things intended to be added. Also, `close()` would be preferable compared to `release()` as it's more abstract and `ReadableBuffer` also uses `close()`.

Reply:
> `close()` would be preferable compared to `release()` as it's more abstract and `ReadableBuffer` also uses `close()`

You seem to be arguing for both sides of this in the same paragraph? :)

> changes like s/release/close/g, casting `int` to `byte`

I just explained these in the preceding comment, they are in fact linked to this PR.

> adding the `len > 0` check

Sure I can remove this if you prefer